### PR TITLE
Project: use a custom template filter to avoid zh languages

### PR DESCRIPTION
**Pull request by humitos**
We keep returning 500 on projects with `zh`.
This another interation on this fix for now.

Sentry https://read-the-docs.sentry.io/issues/5066249292/
Related #370

### CODE REVIEW

**Comment by humitos** [comment]
I couldn't fix the linting regarding inline CSS styles because we need them.

File: readthedocsext/theme/templates/includes/elements/chips/project.html [hunk 1]
```diff
@@ -2,28 +2,32 @@
...
+{% endblock chip_url %}
 
 {% block chip_classes %}
   basic image
 {% endblock chip_classes %}
 
 {% block chip_icon %}
   {% if project.remote_repository %}
-    <img src="{{ project.remote_repository.avatar_url }}" />
+    <img alt="Project avatar" src="{{ project.remote_repository.avatar_url }}" />
```
Comment (agjohnson) [inline readthedocsext/theme/templates/includes/elements/chips/project.html:21]: This needs translation and should mention the project name. I did a version of this in 

- #403 

Note, there are a few cases of this below that need the same fix.

File: readthedocsext/theme/templatetags/ext_theme_tags.py [hunk 2]
```diff
@@ -124,3 +129,29 @@ def get_spam_score(project):
...
+# Simple solution to not supported "zh" language code.
+#
+# When `project.language="zh"` the Django `language_name_local` raises an exception and returns 500.
+# This simple solution checks for this unsupported language code and replace it by "zh-cn" (Simplified Chinese).
+# We need to decide what to do at the DB level still, but at least this approach solves the immediate issue.
+@register.filter
+def readthedocs_language_name(lang_code):
+    if lang_code == "zh":
+        return language_name("zh-cn")
+    return language_name(lang_code)
```
Comment (agjohnson) [inline readthedocsext/theme/templatetags/ext_theme_tags.py:143]: This is what I was thinking of doing originally. It seems like a fine fix for now :+1:

File: readthedocsext/theme/templates/includes/footer.html [hunk 3]
```diff
@@ -136,15 +145,15 @@ <h4 class="ui inverted sub header">Tools</h4>
     {% block footer_sponsors %}
       {% if USE_PROMOS %}
 
-        <div class="ui inverted horizontal divider very padded">
-          Sponsored by
-        </div>
+        <div class="ui inverted horizontal divider very padded">Sponsored by</div>
```
Comment (agjohnson) [inline readthedocsext/theme/templates/includes/footer.html:148]: Guess we need to tune djlint for these, I prefer the original on all of these cases where it's reduced down to a single line. Not a now thing, we should open an issue to track some of the djlint fixes perhaps though.

File: readthedocsext/theme/templates/projects/partials/header.html [hunk 4]
```diff
@@ -119,7 +119,9 @@
               <div class="ui overlapping avatar images">
                 {% for user in project.users.all|slice:":8" %}
                   <a href="{% url "profiles_profile_detail" user.username %}">
-                    <img class="ui image" src="{% gravatar_url user.email 32 %}" />
+                    <img alt="User avatar"
```
Comment (agjohnson) [inline readthedocsext/theme/templates/projects/partials/header.html:122]: Also needs translation and should mention the user.

File: readthedocsext/theme/templates/projects/partials/header.html [hunk 5]
```diff
@@ -138,7 +140,7 @@
               {% if project.remote_repository %}
                 {% with repo=project.remote_repository %}
                   <a href="{{ repo.html_url }}" class="ui basic image nowrap label">
-                    <img src="{{ repo.avatar_url }}" />
+                    <img alt="Project avatar" src="{{ repo.avatar_url }}" />
```
Comment (agjohnson) [inline readthedocsext/theme/templates/projects/partials/header.html:143]: Also needs translation and mention of project name

**Review by agjohnson — APPROVED** [review]
This looks great :+1:

I figured we might need to come back and do this change. It seems like the best approach until we have a database fix.

**Comment by humitos** [comment]
We need a way to ignore these...

```
readthedocsext/theme/templates/projects/partials/header.html
───────────────────────────────────────────────────────────────────────────────
H021 47:10 Inline styles should be avoided. <div class="eight wi
H021 110:10 Inline styles should be avoided. <div class="right al

Linted 2 files, found 2 errors.
```

**Comment by humitos** [comment]
I'll proceed and merge this since the linting issues are not related to this PR. We can fix them in another one.

**Comment by agjohnson** [comment]
You can add a djlint rule. The documentation covers how to exclude rules globally and skip locally in templates.

**Pull request merged**